### PR TITLE
Update `flutter.gradle` AGP to 7.2.0 and bump default NDK version

### PR DESCRIPTION
**Pull request by GaryQian**
Addresses https://github.com/flutter/flutter/issues/105040 as we have since updated the default AGP version to 7.2.0 already.

### CODE REVIEW

**Comment by GaryQian** [comment]
AndroidProject.FD_INTERMEDIATES has been deprecated. We need to find an alternative or bump to 7.1.4 for now.

**Comment by GaryQian** [comment]
It seems that the new AGP simply deprecated the use of FD_INTERMEDIATES as a variable and instead just directly codes in `intermediates` as the directory to use. Thus, this PR now follows that example by using a top level INTERMEDIATES_DIR variable that maps to `intermediates`

File: packages/flutter_tools/gradle/flutter.gradle [hunk 1]
```diff
@@ -752,7 +755,9 @@ class FlutterPlugin implements Plugin<Project> {
      * from the command line.
      *
      * The AGP team said that this issue is fixed in Gradle 7.0, which isn't released at the
-     * time of adding this code. Once released, this can be removed.
+     * time of adding this code. Once released, this can be removed. However, after updating to
+     * AGP/Gradle 7.2.0/7.5, removing this hack still causes build failures. Futher
+     * investigation necessary to remote this.
```
Comment (stuartmorgan) [inline packages/flutter_tools/gradle/flutter.gradle:760]: Is there an issue we can reference here in a TODO?

**Review by stuartmorgan-g — APPROVED** [review]
LGTM with nit

File: packages/flutter_tools/gradle/flutter.gradle (hunk 1 above)
Comment (GaryQian) [reply packages/flutter_tools/gradle/flutter.gradle:760]: Let me file something and add a bit more detail.

**Pull request merged**